romancal/source_catalog/source_catalog_step.py

```diff
@@ -94,35 +95,68 @@ def process(self, input_model):
 
             if segment_img is None:  # no sources found
                 source_catalog_model.source_catalog = Table()
-                return source_catalog_model
-
-            ci_star_thresholds = (self.ci1_star_threshold, self.ci2_star_threshold)
-            catobj = RomanSourceCatalog(
-                model,
-                segment_img,
-                convolved_data,
-                aperture_params,
-                ci_star_thresholds,
-                self.kernel_fwhm,
-                self.fit_psf,
-            )
+            else:
+                ci_star_thresholds = (self.ci1_star_threshold, self.ci2_star_threshold)
+                catobj = RomanSourceCatalog(
+                    model,
+                    segment_img,
+                    convolved_data,
+                    aperture_params,
+                    ci_star_thresholds,
+                    self.kernel_fwhm,
+                    self.fit_psf,
+                )
 
-            # put the resulting catalog in the model
-            source_catalog_model.source_catalog = catobj.catalog
+                # put the resulting catalog in the model
+                source_catalog_model.source_catalog = catobj.catalog
 
             # add back background to data so input model is unchanged
             # (in case of interactive use)
             model.data += bkg.background
 
-            if self.save_results:
-                # NOTE: the source_catalog_model is automatically saved
-                #       if save_results = True
+            # create catalog filename
+            # N.B.: self.save_model will determine whether to use fully qualified path or not
+            output_catalog_name = self.make_output_path(
+                basepath=model.meta.filename, suffix="cat"
+            )
 
-                # save the segmentation map
-                segmentation_model = maker_utils.mk_datamodel(
-                    datamodels.MosaicSegmentationMapModel
-                )
-                segmentation_model.data = segment_img.data.astype(np.uint32)
-                self.save_model(segmentation_model, suffix="segm")
+            if isinstance(model, ImageModel):
+                update_metadata(model, output_catalog_name)
+
+            output_model = model
 
-        return source_catalog_model
+            # always save segmentation image + catalog
+            self.save_base_results(segment_img, source_catalog_model)
+
+            # return the updated model or the source catalog object
+            if self.return_updated_model:
+                # setting the suffix to something else to prevent
+                # step from  overwriting source catalog file with
+                # a datamodel
+                self.suffix = "source_catalog"
```

Comment:
Can we make this something without an underscore?  sourcecatalog?

Reply:
Yes this should match what is in the suffix library, sourcedetection